### PR TITLE
[tensorflow] [test] Fix TF 2.6 dependency check

### PR DESCRIPTION
**Pull request by saimidu**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Fix TF 2.6 dependency check

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [x] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [x] `sagemaker_local_tests = true`

## Label Checklist
- [x] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by saimidu** [comment]
Training tests run successfully. Now running inference tests

File: test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py [hunk 1]
```diff
@@ -205,10 +205,11 @@ def host_setup_for_tensorflow_inference(
             hide=True,
         )
     else:
+        # Installing protobuf at 3.20.* to fix errors in using latest protobuf
         ec2_connection.run(
             (
                 f"{python_invoker} -m pip install --user -qq -U 'tensorflow<={framework_version}' "
-                f" 'tensorflow-serving-api<={framework_version}' "
+                f" 'tensorflow-serving-api<={framework_version}' 'protobuf>=3.20,<3.21'"
```
Comment (tejaschumbalkar) [inline test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:212]: This was handled explicitly for TF2.7 on line 193. Seems it affects TF2.6 as well, we can remove the condition on line 193

Comment (saimidu) [reply test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:212]: Ah, thanks for adding this!!

Comment (saimidu) [reply test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py:212]: I'll wait to confirm that all tests pass, and then remove the pip install at line 193.

Comment (saimidu) [reply test/dlc_tests/ec2/tensorflow/inference/test_tensorflow_inference.py]: @tejaschumbalkar I have removed the protobuf installation that I have added in the non-graviton case, and made the TF 2.7 specific protobuf installation general to all executions of this test.

**Review by waytrue17 — DISMISSED** [review]

**Review by shantanutrip — DISMISSED** [review]
LGTM, lets revert temp chagnes

**Review by shantanutrip — APPROVED** [review]
